### ISSUE TITLE
Why loss and log perplexity is increased during the training?

### ISSUE THREAD

**Issue by amirj** [google/seq2seq#181]
<img width="200" alt="screen shot 2017-04-19 at 10 25 22 am" src="https://cloud.githubusercontent.com/assets/1645137/25195056/afd2b8a6-2551-11e7-948f-151238d62846.png">

**Comment by matthias-samwald** [comment]
@amirj did you eventually have success with training the models based on tutorial data? I have trouble replicating the loss curves shown in the documentation (for the large model), it seems to overfit after reaching a BLEU score of 2,5 to 3...

**Comment by kyleyeung** [comment]
@matthias-samwald Same problem here. I tried all three types of models and different batch sizes, still failed to reproduce the result. My best model started to overfit around a BLEU score of 7~8.

**Comment by matthias-samwald** [comment]
Looking at the text samples generated during training  (which are also saved in the /samples subfolder) as a native German speaker, it seems like ALL of the samples come from the European parliament corpus, but not the other corpora we are supposed to be training on.

The en-de training files are generated by concatenating multiple corpora, and the European parliament corpus is on top. **Could it be that tf-seq2seq is using only a part of the training files (beginning from top) for the training?** That would effectively make the training corpus smaller and could result in the apparent overfitting and low validation BLEU scores we are observing. 

I have no clue why this would happen, though.

**Comment by matthias-samwald** [comment]
So I just searched for the lines shown in the training samples in the training corpus. It seems like among the 10 samples I tested, all of them were drawn from the first few 2000 - 3000 lines of the training document (which has a total of 4,5 M lines). 

So either the samples are drawn very selectively from the top and don't represent the entire corpus, or my tf-seq2seq is working only with a tiny fraction of the corpus. @dennybritz Which could be the case?

**Comment by matthias-samwald** [comment]
Maybe an issue with the queuing in https://github.com/google/seq2seq/blob/master/seq2seq/data/parallel_data_provider.py ?
